src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -42,11 +42,19 @@ std::shared_ptr<PosList> ColumnIsNullTableScanImpl::scan_chunk(const ChunkID chu
 void ColumnIsNullTableScanImpl::_scan_generic_segment(const BaseSegment& segment, const ChunkID chunk_id,
                                                       PosList& matches,
                                                       const std::shared_ptr<const PosList>& position_filter) const {
-  segment_with_iterators_filtered(segment, position_filter, [&](auto it, const auto end) {
-    const auto invert = _predicate_condition == PredicateCondition::IsNotNull;
-    const auto functor = [&](const auto& value) { return invert ^ value.is_null(); };
-
-    _scan_with_iterators<false>(functor, it, end, chunk_id, matches);
+  segment_with_iterators_filtered(segment, position_filter, [&](auto it, [[maybe_unused]] const auto end) {
+    // Don't instantiate this for this for DictionarySegments and ReferenceSegments to save compile time.
+    // ValueSegments are handled in _scan_value_segment()
+    // ReferenceSegments are handled via position_filter
```

Comment:
No, they are not, because `ColumnIsNullTableScanImpl` is not a `AbstractSingleColumnTableScanImpl`. The latter is thus misnamed. Maybe `AbstractDereferencedTableScanImpl` or alike?

Reply:
Why is it not a AbstractSingleColumnTableScanImpl?

Reply:
I removed the apparently unused `position_filter` from the `ColumnIsNullTableScanImpl` because this one's existence was what lead me to believe ReferenceSegments are handled implicitly

Reply:
> Why is it not a AbstractSingleColumnTableScanImpl?

```
  // We need to override scan_chunk because we do not want ReferenceSegments to be resolved (which would remove
  // NullValues from the referencing PosList)
  std::shared_ptr<PosList> scan_chunk(const ChunkID chunk_id) const override;
```

I'd like to rename the `AbstractSingleColumnTableScanImpl` though. The null scan certainly scans a single column, so it would only be logically for it to inherit from that. Options:

* AbstractDereferencedTableScanImpl
* AbstractResolvedReferenceTableScanImpl
* ...?

Reply:
every name will suck, I suppose.... AbstractNonNullTableScan? I guess I could live with AbstractDereferencedTableScanImpl if the class doc makes  is unmistakably clear why this superclass exists and why it is named like that. ColumnIsNullTableScanImpl's docs should likewise be very clear as to why it does NOT inherit from AbstractDereferencedTableScanImpl

Reply:
> There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors.

> ColumnIsNullTableScanImpl's docs should likewise be very clear as to why it does NOT inherit from AbstractDereferencedTableScanImpl

What are you missing in the comment quoted above?

Reply:
Took me longer than a moment to understand why this comment implies that AbstractSingleColumnTableScanImpl is not the base class of ColumnIsNullTableScanImpl.

"// We need to override scan_chunk" implies "We cannot derive from ASCTSI" but that's not an obvious implication.

Reply:
Also I'd really like this to be stated in the class comment of ColumnIsNullTableScanImpl. 

Just come up with something ;)

Reply:
What do you think?